experimental/generation/generator/packages/library/src/dialogGenerator.ts

```diff
@@ -122,6 +125,24 @@ export async function writeFile(path: string, val: string, feedback: Feedback, s
     }
 }
 
+// Return template directories by combining explicit ones with library ones
+export async function templateDirectories(templateDirs?: string[]): Promise<string[]> {
+    // Fully expand all directories
+    templateDirs = resolveDir(templateDirs || [])
```

Comment:
Consider declaring a new variable if the templateDirs is pass-by-value.  Otherwise, you have a side-effect of modifying templateDirs in this function.